packages/syntax/src/Syntax.tsx

```diff
@@ -1,14 +1,56 @@
 import React, { useMemo } from 'react';
 import PropTypes from 'prop-types';
-import { css } from '@leafygreen-ui/emotion';
+import flatten from 'lodash/flatten';
+import { cx, css } from '@leafygreen-ui/emotion';
+import { fontFamilies } from '@leafygreen-ui/tokens';
 // Import from core so we can register the appropriate languages ourselves
 import hljs from 'highlight.js/lib/core';
 import hljsDefineGraphQL from 'highlightjs-graphql';
-import CodeWrapper from './CodeWrapper';
-import { Language, SyntaxProps } from './types';
+import { Language, SyntaxProps, Mode } from './types';
 import { SupportedLanguages, languageParsers } from './languages';
 import { injectGlobalStyles } from './globalStyles';
-import renderingPlugin from './renderingPlugin';
+import renderingPlugin, { TableContent } from './renderingPlugin';
+import { SyntaxContext } from './SyntaxContext';
+
+export function expandRangeTuple(tuple: [number, number]): Array<number> {
+  const [lower, upper] = [...tuple]
+    .map(bound => {
+      // Make sure passing infinity doesn't freeze the browser
+      if (bound === Infinity) {
```

Comment:
I think this is overly-specific and we should clamp any number that's too large, e.g. i.e. `bound = Math.min(bound, 2000)`

Reply:
I was avoiding this simply because I'm not a fan of arbitrary limitations. Highlighting 2000 lines should work, even if the use-case isn't really there. Is there a reason you see being so specific would cause issues?

Reply:
As-is, the Infinity case is exceptionally unlikely.

Reply:
it just makes the interface of the function more complex than necessary. it's simpler to say
"this function returns an array of numbers within the range provided, bounded by 0 and 2000" than it is to say "this function returns an array of numbers within the range provided, unless a bound is Infinity or -Infinity in which it will be replaced by 2000 and -2000 respectively"

Reply:
I don't feel that strongly about this. I'll change it.

Reply:
Actually I just realized, clamping the bounds to 2000 means that even if the tuple is `[1999, 2001]`, it would output `[1999, 2000]`. If we ever add a lines-start sort of property where they could start line numbering at `2001`, this would effectively disable line highlighting. I feel a bit more strongly about this now given that. We could make sure the upper and lower bound difference is than 2000, but that's less elegant of a solution for the `Infinity` problem.